src/dac.rs

```diff
@@ -0,0 +1,279 @@
+///! Stabilizer DAC management interface
+///!
+///! The Stabilizer DAC utilize a DMA channel to generate output updates.  A timer channel is
+///! configured to generate a DMA write into the SPI TXFIFO, which initiates a SPI transfer and
+///! results in DAC update for both channels.
+use super::{
+    hal, sampling_timer, DMAReq, DmaConfig, MemoryToPeripheral, TargetAddress,
+    Transfer, SAMPLE_BUFFER_SIZE,
+};
+
+// The following global buffers are used for the DAC code DMA transfers. Two buffers are used for
+// each transfer in a ping-pong buffer configuration (one is being prepared while the other is being
+// processed). Note that the contents of AXI SRAM is uninitialized, so the buffer contents on
+// startup are undefined.
```

Comment:
Should we explicitly initialize the DAC buffers to some "safe" value? I'm thinking the DACs will spit out garbage when they are first turned on as the DMA reads from the very first buffer.

Reply:
The DAC transfers aren't enabled at first. They are only started after the first `next_data()` call, so there's no uninitialized data ever written to the DACs.